fix-version-gaps/fixGaps.ts

```diff
@@ -0,0 +1,96 @@
+import { Octokit } from '@octokit/rest';
+import { PullRequest } from '@octokit/webhooks-definitions/schema';
+import { ConfigOptions } from 'backport';
+
+export function getLowestVersionsOnPr(pr: PullRequest) {
+  const lowestVersionsOnPr: Record<string, string> = {};
+  for (const label of pr.labels) {
+    const matches = label.name.match(/^v([0-9.]+)$/);
+    if (matches) {
+      const [major, minor] = matches[1].split('.');
+
+      if (!lowestVersionsOnPr[major]) {
+        lowestVersionsOnPr[major] = minor;
+      } else if (parseInt(lowestVersionsOnPr[major], 10) > parseInt(minor, 10)) {
+        lowestVersionsOnPr[major] = minor;
+      }
+    }
+  }
+
+  return lowestVersionsOnPr;
+}
+
+export function getVersionsFromBackportConfig(config: ConfigOptions) {
+  const highestVersions = [];
+
+  for (const label in config.branchLabelMapping) {
+    const matches = label.match(/^\^v([0-9.]+)\$$/);
+    if (matches) {
+      const version = matches[1];
+      highestVersions.push(version);
+    }
+  }
+
+  return highestVersions;
+}
+
+export function getVersionLabelsToAdd(config: ConfigOptions, pr: PullRequest) {
+  const versionsFromBackportConfig = getVersionsFromBackportConfig(config);
+  const lowestVersionsOnPr = getLowestVersionsOnPr(pr);
+  const allLabels = pr.labels.map((label) => label.name);
+
+  const versionLabelsToAdd = [];
+
+  for (const version of versionsFromBackportConfig) {
+    const [major, minor] = version.split('.');
+    const nextVersion = parseInt(lowestVersionsOnPr[major], 10) + 1;
+    for (let i = nextVersion; i <= parseInt(minor, 10); i++) {
+      const label = `v${major}.${i}.0`;
```

Comment:
I feel like defaulting the patch version to 0 here is enough to make sure that PRs are being backported to the right branch, but it would be ideal if they were using the correct patch number too. Can't think of any way to make that work easily though.

Reply:
Yeah, I agree, but we don't have that data anywhere that I can think of at the moment. We really need to start tracking it though, the need for it comes up all the time.